opencog/atoms/base/Atom.h

```diff
@@ -224,6 +225,14 @@ class Atom
     virtual ~Atom();
     virtual bool is_atom() const { return true; }
 
+    /** Basic predicate */
+    bool is_type(Type t, bool subclass = true) const
+    {
```

Comment:
why did you move this from protoAtom to Atom? I think it needs to stay in protoatom, as one needs to be able to query the type of values, and not just atoms.